src/ui/Distance/DistanceResources.ts

```diff
@@ -305,30 +302,19 @@ export class DistanceResources extends Component {
     return providers;
   }
 
-  private tryGetPositionFromProviders(providers: IGeolocationPositionProvider[]): Promise<IGeolocationPosition> {
-    return new Promise<IGeolocationPosition>((resolve, reject) => {
-      const tryNextProvider = () => {
-        if (providers.length > 0) {
-          const provider = providers.shift();
-          provider
-            .getPosition()
-            .then(position => {
-              if (!!position.latitude && !!position.longitude) {
-                resolve(position);
-              } else {
-                tryNextProvider();
-              }
-            })
-            .catch(error => {
-              this.logger.warn('An error occurred while trying to resolve the position within a position provider.', error);
-              tryNextProvider();
-            });
-        } else {
-          resolve();
+  private async tryGetPositionFromProviders(providers: IGeolocationPositionProvider[]): Promise<IGeolocationPosition> {
+    while (providers.length > 0) {
+      const provider = providers.shift();
+      try {
+        const position = await provider.getPosition();
+        if (!!position.latitude && !!position.longitude) {
+          return position;
         }
-      };
-      tryNextProvider();
-    });
+      } catch (error) {
+        this.logger.warn('An error occurred while trying to resolve the position within a position provider.', error);
+      }
+    }
+    return;
```

Comment:
Maybe it would be better to throw here instead of returning `undefined` and needing to always check the return value.

Reply:
Okay, makes sense 👍